### PR TITLE
adding externals as a config parameter

### PR DESCRIPTION
**Pull request by BhuvaneshwaranR**
solution for a [issue ](https://github.com/AnomalyInnovations/serverless-bundle/issues/50). @jayair could you verify and do the needful at the earilest.

### CODE REVIEW

File: src/webpack.config.js [hunk 1]
```diff
@@ -135,8 +136,8 @@ module.exports = ignoreWarmupPlugin({
   // Disable verbose logs
   stats: ENABLE_STATS ? "normal" : "errors-only",
   devtool: ENABLE_SOURCE_MAPS ? "source-map" : false,
-  // Exclude "aws-sdk" since it's a built-in package
-  externals: ["aws-sdk", "knex", "sharp"],
+  // Exclude "aws-sdk" since it's a built-in package and some other packages
+  externals: !config.options.externals ?  externals : externals.concat(config.options.externals),
```
[the remark this answers is not in the document]
Comment (BhuvaneshwaranR) [reply src/webpack.config.js]: Initialize value of `config.options.externals` is null. So
`externals: ["aws-sdk", "knex", "sharp"].concat(config.options.externals)` will be  `["aws-sdk", "knex", "sharp", null]`. It's better to add condition and perform concationation.

**Comment by jayair** [comment]
Thanks for this! Based on the other thread (https://github.com/AnomalyInnovations/serverless-bundle/issues/81), do we need to update the default list with the package that's causing the issue?

**Comment by BhuvaneshwaranR** [comment]
@jayair We don't want to update the default list. We have to provide an option to ignore packages through configuration.

**Comment by thibaultdalban** [comment]
This PR introduces a must-have option, so thank you @BhuvaneshwaranR :+1: 
IMHO, it's unconventional to merge default values with the custom ones. Usually, when an option is specified, it's to override the default value.
I suggest adding a note to the README to:
- Let the users know about this new option :tada: 
- Avoid the confusion, with a description of the behaviour.

File: src/webpack.config.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (thibaultdalban) [reply src/webpack.config.js:140]: Or you could initialize the `externals` with `[]` and then just do:
```js
externals: ["aws-sdk", "knex", "sharp"].concat(config.options.externals)
```
as suggested by @domnantas.

**Comment by jayair** [comment]
@thibaultdalban Can you expand on that? Do you want the externals option to override what the plugin is setting? Or do you want to concat it like the comment you left in the code review?

**Comment by thibaultdalban** [comment]
@jayair from my point of view, it's less confusing to let the externals option entirely overrides what the plugin is setting.

**Comment by suryacaprice** [comment]
how do i use this externals feature

File: src/webpack.config.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (BhuvaneshwaranR) [reply src/webpack.config.js:140]: Yes, We can do that.

**Comment by jayair** [comment]
@BhuvaneshwaranR do you want to make the proposed changes and add to the PR?

**Comment by BhuvaneshwaranR** [comment]
@jayair  Added suggested changes and readme document. Please merge the changes at the earilest.

**Comment by jayair** [comment]
Awesome!

@BhuvaneshwaranR  @thibaultdalban Thanks for contributing.

It's been released.

https://github.com/AnomalyInnovations/serverless-bundle/releases/tag/v1.7.0

**Pull request merged**